File: idealingua/idealingua-runtime-rpc-c-sharp/src/main/resources/runtime/csharp/IRT/Transport/Authorization/AuthApiKey.cs

using System;

namespace IRT.Transport.Authorization {
    class AuthApiKey: AuthMethod {
        public string ApiKey;

        public AuthApiKey(string apiKey = null) {
            ApiKey = apiKey;
        }

        public override bool FromValue(string value) {
            var lower = value.ToLower();
            if (value.StartsWith("api-key ", StringComparison.Ordinal)) {
                ApiKey = value.Substring(8);
                return true;
            }

            if (value.StartsWith("apikey ", StringComparison.Ordinal)) {
                ApiKey = value.Substring(7);
                return true;
            }

            return false;
        }

        public override string ToValue() {
            return "Api-Key " + ApiKey;
        }
    }
}

File: idealingua-v1/idealingua-v1-runtime-rpc-typescript/src/main/resources/runtime/typescript/irt/transport/auth/auth.apikey.ts
import { AuthMethod } from './auth.method';

export class AuthApiKey extends AuthMethod {
    public apiKey: string;
    constructor(apiKey: string) {
        super();
        this.apiKey = apiKey;
    }

    public fromValue(value: string): Error {
        if (value.toLowerCase().indexOf('api-key ') === 0) {
            this.apiKey = value.substr(8);
            return undefined;
        }

        if (value.toLowerCase().indexOf('apikey ') === 0) {
            this.apiKey = value.substr(7);
            return undefined;
        }

        return new Error('api key authorization must start with ApiKey, got ' + value)
    }

    public toValue(): string {
        return 'Api-Key ' + this.apiKey;
    }
}
